src/RouteLockedWorkspacesManager.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Drupal\workspaces_route_lock;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\workspaces\Entity\Workspace;
+use Drupal\workspaces\WorkspaceInterface;
+use Drupal\workspaces\WorkspaceManagerInterface;
+use Drupal\workspaces_route_lock\Entity\WorkspacesRouteLock;
+
+class RouteLockedWorkspacesManager implements WorkspaceManagerInterface {
+
+  /**
+   * The underlying manager service.
+   *
+   * @var \Drupal\workspaces\WorkspaceManagerInterface
+   */
+  protected $workspacesManager;
+
+  /**
+   * The current route match.
+   *
+   * @var RouteMatchInterface
+   */
+  protected $currentRouteMatch;
+
+  /**
+   * The cache backend.
+   *
+   * @var CacheBackendInterface
+   */
+  protected $cache;
+
+  /**
+   * WorkspacesModerationManager constructor.
+   *
+   * @param \Drupal\workspaces\WorkspaceManagerInterface $workspaceManager
+   *   The underlying manager service.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $currentRouteMatch
+   *   The current route match.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
+   */
+  public function __construct(WorkspaceManagerInterface $workspaceManager, RouteMatchInterface $currentRouteMatch, CacheBackendInterface $cache) {
+    $this->workspacesManager = $workspaceManager;
+    $this->currentRouteMatch = $currentRouteMatch;
+    $this->cache = $cache;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public function getActiveWorkspace() {
+    $activeWorkspace = $this->workspacesManager->getActiveWorkspace();
+    $routeName = $this->currentRouteMatch->getRouteName();
+
+    if ($routeName) {
+      $cid = "workspaces_route_lock:$routeName";
+      $cacheItem = $this->cache->get($cid);
+
+      if (!$cacheItem) {
+        /** @var \Drupal\workspaces_route_lock\Entity\WorkspacesRouteLockInterface $routeLock */
+        foreach (WorkspacesRouteLock::loadMultiple() as $routeLock) {
+          if ($routeLock->appliesToRoute($routeName)) {
+            if (!$routeLock->workspaceIsAllowed($this->getWorkspaceId($activeWorkspace))) {
```

Comment:
`WorkspaceManagerInterface::getActiveWorkspace` always returns a workspace. The interface documentation cannot lie :)

So I think there is no need for `getWorkspaceId` method. Also the method is a bit confusing. I mean I could not guess its behavior from its name.

Reply:
In 8.9 it can return null.